cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestLocationGroupController.java

```diff
@@ -59,18 +64,95 @@ void testRetrieveLocationGroup() throws IOException {
+        mockHttpServer.start();
+        LocationGroupEndpointInput input = new LocationGroupEndpointInput()
+            .includeAssigned(true)
+            .officeId("CWMS");
+        List<LocationGroup> locationGroups = new LocationGroupController().retrieveLocationGroups(buildConnectionInfo(), input);
+
+        assertEquals(3, locationGroups.size());
+        LocationGroup locationGroup = locationGroups.get(1);
+        assertEquals("CWMS", locationGroup.getOfficeId());
+        assertEquals("CWMS Legacy Naming", locationGroup.getId());
+        assertEquals("Alias group for legacy location IDs", locationGroup.getDescription());
+        LocationCategory locationCategory = locationGroup.getLocationCategory();
+        assertEquals("Agency Aliases", locationCategory.getId());
+        assertEquals("CWMS", locationCategory.getOfficeId());
+        assertEquals("Location aliases for other agencies", locationCategory.getDescription());
+
+        List<AssignedLocation> assignedLocations = locationGroup.getAssignedLocations();
+        AssignedLocation assignedLocation = assignedLocations.get(0);
+        assertEquals("Deleted TS ID", assignedLocation.getLocationId());
+        assertEquals("CWMS", assignedLocation.getOfficeId());
+        assertEquals("Deleted TS ID", assignedLocation.getAliasId());
+    }
+
+    @Test
+    void testRetrieveLocationGroupsNoAssignedLocations() throws IOException {
+        String resource = "radar/v1/json/location_groups_noassignedlocs.json";
+        URL resourceUrl = getClass().getClassLoader().getResource(resource);
+        if (resourceUrl == null) {
+            throw new IOException("Failed to get resource: " + resource);
+        }
+        Path path = new File(resourceUrl.getFile()).toPath();
+        String collect = String.join("\n", Files.readAllLines(path));
+        mockHttpServer.enqueue(collect);
+        mockHttpServer.start();
+        LocationGroupEndpointInput input = new LocationGroupEndpointInput()
+            .includeAssigned(false)
+            .officeId("CWMS");
+        List<LocationGroup> locationGroups = new LocationGroupController().retrieveLocationGroups(buildConnectionInfo(), input);
+
+        assertEquals(823, locationGroups.size());
+        LocationGroup locationGroup = locationGroups.get(1);
+        assertEquals("SWT", locationGroup.getOfficeId());
+        assertEquals("CWMS Standard Naming", locationGroup.getId());
+        assertEquals("Alias group for location IDs that conform to CWMS Standard Naming Convention", locationGroup.getDescription());
+        LocationCategory locationCategory = locationGroup.getLocationCategory();
+        assertEquals("Agency Aliases", locationCategory.getId());
+        assertEquals("CWMS", locationCategory.getOfficeId());
+        assertEquals("Location aliases for other agencies", locationCategory.getDescription());
+
+        List<AssignedLocation> assignedLocations = locationGroup.getAssignedLocations();
```

Comment:
not an empty collection>?

Reply:
initializing collections to an empty collection is a feature in the next version of the swagger generator, so for now it's null.